packages/repository/src/decorators/metadata.ts

```diff
@@ -0,0 +1,39 @@
+import {InspectionOptions, MetadataInspector} from '@loopback/context';
+import {MODEL_KEY, MODEL_PROPERTIES_KEY} from '../';
+import {ModelDefinition, PropertyDefinition} from '../../index';
+
+export class ModelMetadataHelper {
+  /**
+   * A utility function to simplify retrieving metadata from a target model and
+   * its properties.
+   * @param target The class from which to retrieve metadata.
+   * @param options An options object for the MetadataInspector to customize
+   * the output of the metadata retrieval functions.
+   */
+  static getModelMetadata(target: Function, options?: InspectionOptions) {
+    const classDef = MetadataInspector.getClassMetadata(
```

Comment:
Do we want to cache the result on the target? See an example at https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/router/metadata.ts#L177

Reply:
I've added some code to alter the MODEL_KEY/target metadata if there aren't any properties defined on first call.